recipes_source/recipes/profiler.py

```diff
@@ -0,0 +1,203 @@
+
+with profiler.profile(profile_memory=True, record_shapes=True) as prof:
+    model(inputs)
+
+print(prof.key_averages().table(sort_by="self_cpu_memory_usage", row_limit=10))
+
+# (omitting some columns)
+# ---------------------------  ---------------  ---------------  ---------------
+# Name                         CPU Mem          Self CPU Mem     Number of Calls
+# ---------------------------  ---------------  ---------------  ---------------
+# empty                        94.79 Mb         94.79 Mb         123
+# resize_                      11.48 Mb         11.48 Mb         2
+# addmm                        19.53 Kb         19.53 Kb         1
+# empty_strided                4 b              4 b              1
+# conv2d                       47.37 Mb         0 b              20
+# ---------------------------  ---------------  ---------------  ---------------
+
+print(prof.key_averages().table(sort_by="cpu_memory_usage", row_limit=10))
+
+# (omitting some columns)
+# ---------------------------  ---------------  ---------------  ---------------
+# Name                         CPU Mem          Self CPU Mem     Number of Calls
+# ---------------------------  ---------------  ---------------  ---------------
+# empty                        94.79 Mb         94.79 Mb         123
+# batch_norm                   47.41 Mb         0 b              20
+# _batch_norm_impl_index       47.41 Mb         0 b              20
+# native_batch_norm            47.41 Mb         0 b              20
+# conv2d                       47.37 Mb         0 b              20
+# convolution                  47.37 Mb         0 b              20
+# _convolution                 47.37 Mb         0 b              20
+# mkldnn_convolution           47.37 Mb         0 b              20
+# empty_like                   47.37 Mb         0 b              20
+# max_pool2d                   11.48 Mb         0 b              1
+# max_pool2d_with_indices      11.48 Mb         0 b              1
+# resize_                      11.48 Mb         11.48 Mb         2
+# addmm                        19.53 Kb         19.53 Kb         1
+# adaptive_avg_pool2d          10.00 Kb         0 b              1
+# mean                         10.00 Kb         0 b              1
+# ---------------------------  ---------------  ---------------  ---------------
+
+######################################################################
+# 5. Using tracing functionality
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# Profiling results can be outputted as a .json trace file:
+
+with profiler.profile() as prof:
+    with profiler.record_function("model_inference"):
+        model(inputs)
+
+prof.export_chrome_trace("trace.json")
+
+######################################################################
+# User can examine the sequence of profiled operators after loading the trace file
+# in Chrome (``chrome://tracing``):
+#
+# .. image:: ../../_static/img/trace_img.png
+#    :scale: 25 %
```

Comment:
Are there additional resources for users to learn more or follow-up topics (e.g. docs, tutorials) we can add as resources?

Reply:
adding a link to TensorBoard viz